test/js/node/net/node-net-server.test.ts

```diff
@@ -463,34 +463,6 @@ describe("net.createServer events", () => {
       });
   });
 
-  it("should call error", done => {
-    const { mustCall, mustNotCall } = createCallCheckCtx(done);
-
-    let timeout: Timer;
-    const server: Server = createServer();
-
-    const closeAndFail = () => {
-      clearTimeout(timeout);
-      server.close();
-      mustNotCall("error not called")();
-    };
-
-    //should be faster than 100ms
-    timeout = setTimeout(closeAndFail, 100);
-
-    server
-      .on(
-        "error",
-        mustCall(err => {
-          server.close();
-          clearTimeout(timeout);
-          expect(err).toBeDefined();
-          done();
-        }),
-      )
-      .listen(123456);
-  });
-
```

Comment:
Node does not exhibit this behaviour 

This image shows:
- Current Bun (handles error in error listener, which is wrong)
- Node 23 (which handles in the catch)
- My local debug build of bun (which also handles in the catch, mimicking Node's behaviour)

![CleanShot 2025-05-19 at 13 44 02@2x](https://github.com/user-attachments/assets/56437636-1cfc-4753-a86e-783323726838)